lib/benchee/formatter.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Benchee.Formatter do
+  @moduledoc """
+  Defines a behaviour for formatters in Benchee, and also defines functions to
+  handle invoking that defined behavior.
+  """
+
+  alias Benchee.{Suite, Utility.Parallel}
+
+  @callback output(Suite.t) :: Suite.t
+  @callback format(Suite.t) :: any
+  @callback write(any) :: :ok | {:error, String.t}
+
+  @doc """
+  Invokes `format/1` and `write/1` as defined by the `Benchee.Formatter`
+  behaviour. The output for all formatters are generated in parallel, and then
+  the results of that formatting are written in sequence.
+  """
+  @spec parallel_output(Suite.t, [module]) :: Suite.t
+  def parallel_output(suite, modules) do
+    modules
+    |> Parallel.map(fn(module) ->
+         {module, module.format(suite)}
+       end)
```

Comment:
doesn't this one fit in one line? (multi line in a pipe always looks a bit subptimal to me compared to all one liners)

Reply:
Yep, it does fit. Not sure why I did that, to be honest!